projects/storefrontlib/cms-structure/seo/media-preconnect.service.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import { PageMetaLinkService } from './page-meta-link.service';
+import { WindowRef, FeatureToggles } from '@spartacus/core';
+import { MediaService } from '../../shared/components/media/media.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MediaPreconnectService {
+  protected pageMetaLinkService = inject(PageMetaLinkService);
+  protected mediaService = inject(MediaService);
+  private featureToggles = inject(FeatureToggles);
+  private windowRef = inject(WindowRef);
```

Comment:
Regular dependencies (besides FeatureToggles) should be `protected` so customers can use them in their extended classes:

```suggestion
  protected windowRef = inject(WindowRef);
```